docs/userguide/04_bespoke-executor.md

```diff
@@ -0,0 +1,141 @@
+Python code using the [`BespokeExecutor`] class, or from the CLI.
+
+[BespokeExecutor]: openff.bespokefit.executor.executor.BespokeExecutor
+[`BespokeExecutor`]: openff.bespokefit.executor.executor.BespokeExecutor
+
+## From Python
+
+Configure the `BespokeExecutor` instance by specifying the number of worker
+threads to prepare for each category of tasks. For example, a reasonable
+configuration for a quad core workstation might be:
+
+```python
+from openff.bespokefit.executor import BespokeExecutor
+
+executor = BespokeExecutor(
+    n_fragmenter_workers = 1,
+    n_qc_compute_workers = 1,
+    n_optimizer_workers = 2,
+)
+```
+
+Note that `n_qc_compute_workers` only controls `BespokeExecutor`, and the
+quantum chemistry code may spin up threads of its own. As a result, using a
+large value for `n_qc_compute_workers` may end up over-subscribing the
+available cores, even if only one workflow is submitted. If a value greater
+than 1 is provided, each worker can work on its own fragment of the target
+molecule.
+
+The resulting executor is a context manager that accepts workflows via its
+`submit()` method. Recall that workflows are instances of the
+[`BespokeOptimizationSchema`] class, which are created by the
+[`optimization_schema_from_molecule()`] method on [`BespokeWorkflowFactory`]. As
+soon as the context manager ends, the `BespokeExecutor` instance is closed,
+terminating any running jobs. To complete a job, we need to use the
+`wait_until_complete()` function, which blocks progress in the script until it
+can return a result. This is also the easiest way to get results out of the
+executor:
+
+```python
+//  from openff.bespokefit.executor import BespokeExecutor
+//  from openff.bespokefit.workflows import BespokeWorkflowFactory
+//  from openff.toolkit.topology import Molecule 
+//  factory = BespokeWorkflowFactory()
+//  target_molecule = Molecule.from_smiles("C(C(=O)O)N") # Glycine
+//  workflow = factory.optimization_schema_from_molecule(
+//      molecule=target_molecule,
+//  )
+//  
+    with BespokeExecutor(1, 1, 2) as executor:
+        task = executor.submit(workflow)
+        result = wait_until_complete(task.id)
+```
+
+These functions respectively correspond to a HTTP POST and GET request, and they
+return the appropriate [POST] and [GET] response types. `wait_until_complete()`
+returns the result of a task even if the task has already finished when the
+function is executed, so using the context manager with multiple workflows is
+simple. Given an iterator of workflow schemas named `workflows`:
+
+```python
+//  from openff.bespokefit.executor import BespokeExecutor
+//  from openff.bespokefit.workflows import BespokeWorkflowFactory
+//  from openff.toolkit.topology import Molecule
+//  target_molecule = Molecule.from_smiles("C(C(=O)O)N") # Glycine
+//  workflow = factory.optimization_schema_from_molecule(
+//      molecule=target_molecule,
+//  )
```

Comment:
These lines may not actually be getting used

Reply:
Good point! Removed